### PR TITLE
Add snapping functionality

### PR DESCRIPTION
**Pull request by lisawink**
Have added a new snapping function to gap.py and accompanying tests to test_gap.py. This allows geometries to be snapped together if they are within a threshold distance from each other and not already touching. Requires geopandas==1.0.0a1 which I have added to the env 312-latest.yaml

### CODE REVIEW

File: geoplanar/gap.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
 #!/usr/bin/env python3
 #
+import pandas as pd
+import numpy as np
 import geopandas
 import shapely
 from packaging.version import Version
 
-__all__ = ["gaps", "fill_gaps"]
+__all__ = ["gaps", "fill_gaps", "snap", "_snap"]
```
Comment (sjsrey) [inline geoplanar/gap.py:9]: If `_snap` is private, should it be added to the package namespace?

**Review by sjsrey — COMMENTED** [review]
Thanks for this!

Would it be possible to add a notebook that demonstrates a small example of this functionality?

File: geoplanar/tests/test_gap.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def test_fill_gaps_gaps_df(self):
         filled = fill_gaps(self.gdf, gaps_df)
         assert_equal(filled.area, numpy.array([104, 32]))
 
-
+@pytest.mark.skipif(geopandas.__version__ != "1.0.0-alpha1", reason="requires geopandas alpha release")
```
Comment (martinfleis) [inline geoplanar/tests/test_gap.py:40]: Requires 1.0 or newer. So 

```py
from packaging.version import Version

@pytest.mark.skipif(Version(geopandas.__version_) >= Version("1.0.0dev"), reason="requires geopandas 1.0")
```

I am also not sure if skipif works on the whole class.

Comment (sjsrey) [reply geoplanar/tests/test_gap.py:40]: I think the skip is working as the tests on earlier versions of gp are [now passing](https://github.com/sjsrey/geoplanar/actions/runs/8968811423).

Comment (lisawink) [reply geoplanar/tests/test_gap.py:40]: Okay will change this, thank you

**Comment by lisawink** [comment]
Found an issue in the snapping function when I was creating an example notebook, so better not to merge until this is fixed and the example notebook is ready

File: geoplanar/gap.py [hunk 3]
```diff
@@ -224,13 +224,21 @@ def snap(geometry, threshold=0.5):
         pairs_to_snap = nearby_not_overlap[~duplicated]
     
         new_geoms = []
+        previous_geom=0
+        snapped_geom=0
```
Comment (martinfleis) [inline geoplanar/gap.py:228]:
```suggestion
        previous_geom=None
        snapped_geom=None
```

I know it won't make a huge difference but it si easier to understand that the variable is not meant to be an integer.

**Review by martinfleis — APPROVED** [review]
failure in the dev env is not related. pandas just broke geopandas. will sort it elsewhere.

**Pull request merged**